download/retry_test.go

```diff
@@ -0,0 +1,101 @@
+package download_test
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	"github.com/Azure/custom-script-extension-linux/download"
+	"github.com/ahmetalpbalkan/go-httpbin"
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	// how much we sleep between retries
+	sleepSchedule = []time.Duration{
+		5 * time.Second,
+		10 * time.Second,
+		20 * time.Second,
+		40 * time.Second,
+		80 * time.Second,
+		160 * time.Second}
+)
+
+func TestActualSleep_actuallySleeps(t *testing.T) {
+	s := time.Now()
```

Comment:
this is because somebody perhaps changes `var ActualSleep` with some mock which doesn't actually sleep like wall clock. it'll detect that side-effect.